src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -160,6 +160,13 @@ def load_arguments(self, _):
     with self.argument_context('feature list') as c:
         c.argument('resource_provider_namespace', options_list='--namespace', required=False, help=_PROVIDER_HELP_TEXT)
 
+    with self.argument_context('feature registration') as c:
+        c.argument('resource_provider_namespace', options_list='--namespace', required=True, help=_PROVIDER_HELP_TEXT)
+        c.argument('feature_name', options_list=['--name', '-n'], help='the feature name')
+
+    with self.argument_context('feature registration list') as c:
+        c.argument('resource_provider_namespace', options_list='--namespace', required=False, help=_PROVIDER_HELP_TEXT)
```

Comment:
`required=False` does not need to be added, `False` is the default value for `required`

Reply:
removed

Reply:
no, I have to add it back as the CI job failed " There is a violation:
 "az feature registration list" is not a valid command.
 the following arguments are required: --namespace."

Reply:
seems the default value is True other than false

Reply:
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#keyword-argument-reference
> required - See https://docs.python.org/3/library/argparse.html#required. Note that this value is inferred from the function signature depending on whether or not the parameter has a default value. If specified, this will override that value.

Reply:
This CI issue is caused by you defining the same parameter in the parent command of the upper level and setting it to `required=True`.
Please define the parameter `resource_provider_namespace` separately for `feature registration list` and `feature registration create/delete`, not in the parent command

Reply:
no, I cannot remove the 'requried=false' as there is top level thing 'feature' which contains 'required=True'.

Even after I remove 'feature registration', the default value will stll be true as the 'feature' has 'required=True'

Reply:
OK